CONTRIBUTING.md

```diff
@@ -0,0 +1,82 @@
+- [GitHub Issues](#gh-issues)
+- [Pull Requests](#prs)
+- [Creating a Pull Request](#creating-a-pr)
+- [Best Practices/Guidelines](#best-practices)
+
+#### Other links
+- [redox-os.org](http://redox-os.org)
+
+<a name="slack"/>
+## Slack Chat
+The quickest and most open way to communicate with the Redox team is with Slack. Currently, the only way to join our slack team is by sending an email to [info@redox-os.org](mailto:info@redox-os.org), which might take a little while, since it's not automated. We're currently working on an easier way to do this, but this is the most convenient way right now.
+
+<a name="gh-issues"/>
+## GitHub Issues
+A bit more formal way of communication with fellow Redox devs, but a little less quick and convienent like the Slack chat (unless of course you aren't in it yet, which if you're going to be involved in this project really at all, it is recommended that you request to join). These are for more specific topics, simply put, issues try to state something more than ask.
+
+<a name="prs"/>
+## Pull Requests
+It's completely fine to just submit a small pull request without first making an issue or something, but if it's a big change that will require a lot of planning and reviewing, it's best you start with writing an issue first.
+
+<a name="creating-a-pr"/>
+## Creating a Pull Request in 10 Steps
+0. Fork the repository
+0. Clone the original repository to your local PC using one of the following commands based on the protocol you are using:
+HTTPS:`git clone https://github.com/redox-os/redox.git`
+SSH:`git clone git@github.com:redox-os/redox.git --origin upstream --recursive`
+Use HTTPS if you don't know which one to use.
+0. Add your fork with
+HTTPS:`git remote add origin https://github.com/your-username/redox.git`
+SSH:`git remote add origin git@github.com:your-username/redox.git --origin upstream --recursive`
+0. Optionally create a separate branch
+0. Make changes
+0. Commit (`git add . --all; git commit -m "my commit"`)
+0. Test your changes with `make qemu` or `make virtualbox` (you might have to use `make qemu_no_kvm`)
+0. Push to your fork (`git push origin my-branch`)
+0. Create a pull request
+0. Describe your changes
+0. Submit!
+
+<a name="best-practices"/>
+## Best Practices/Guidelines
+<!-- TODO add this section to the index/TOC -->
+
+#### General
+
+- When trying to access a slice, always always ALWAYS use the `common::GetSlice` trait and use the `.get_slice()` method to get a slice without causing the kernel to panic. The problem with slicing in regular Rust, e.g. `foo[a..b]`, is that if someone tries to access with a range that is out of bounds of an array/string/slice, it will cause a panic at runtime, as a safety measure. Same thing when accessing an element. Always use `foo.get(n)` instead of `foo[n]` and try to cover for the possibility of `Option::None`. Doing the regular way may work fine for applications, but never in the kernel. No possible panics should ever exist in kernel space.
```

Comment:
You can change the `always always ALWAYS` to `you must always` just to be a bit more formal. You could also bold the word "always" to stress the necessity.